### DISCUSSION TITLE
哔哩哔哩, 暂停的时候快进 不要重新继续播放视频.

### DISCUSSION THREAD

**Discussion by HornyPrivateGamer** [the1812/Bilibili-Evolved#4979] [功能建议 / Ideas]
就感觉看的时候想暂停然后想快进下，总是重新开始播放,能不能做得像 YouTube 一样
现在基本上所有的播放器都是暂停后，如果快进的话视频是不会重新开始播放的, MPV, Youtube. 
谢谢，就这样

而且对的，我就是24号那个捐钱的那个人 :)

如果楼主同意做这个功能的话，我立刻可以给100， 事成之后再给150
@the1812

**Comment by HornyPrivateGamer** [comment]
Ha lo Anyone ?

**Comment by the1812** [comment]
脚本调的也是 b 站提供的播放器 API (https://github.com/the1812/Bilibili-Evolved/discussions/4341)
以前调用 `seek` 的时候第二个参数就是是否暂停, 那时候是可以做到的, 不过现在这个参数改成一个叫 `initiator` 的东西了, 没法控制了

附 b 站播放器 `seek` 的代码:
```js
t.prototype.seek = function (e, t) {
  return (0, n.mG)(this, void 0, void 0, function () {
    var r,
      i,
      s,
      u,
      c,
      l,
      d,
      f,
      p,
      h,
      y = this
    return (0, n.Jh)(this, function (m) {
      switch (m.label) {
        case 0:
          if (
            ((r = this.rootPlayer.disabled.seek || this.rootPlayer.hasFeature('noSeek')),
            this.log.i(
              '[seek]',
              ee(
                ve ||
                  (ve = (0, n.cy)(
                    ['params:', ',alt:', ',disabled:', ''],
                    ['params:', ',alt:', ',disabled:', ''],
                  )),
                e,
                t,
                r,
              ),
            ),
            (i = (null == t ? void 0 : t.initiator) || null),
            (u = (s =
              'number' == typeof e
                ? {
                    value: e,
                  }
                : e).isOffset
              ? this.getCurrentTime() + s.value
              : s.value),
            (c = (0, C.Z)(u, 0, this.getDuration() - 0.1)),
            this.patchAdvStore.hasValidAdv)
          ) {
            if (
              ((l = (0, C.Z)(e, 0, this.getDuration() - 0.1)),
              (d = this.patchAdvStore.getSeekDecision(l)),
              (f = d.isSameEp),
              (p = d.t),
              (h = d.index),
              !f)
            )
              return (
                this.patchAdvStore.setState({
                  index: h,
                }),
                [2, this.patchAdvStore.patchSwitch(p)]
              )
            c = p
          }
          if (r) throw new Error('Seek Disabled')
          if (isNaN(c)) throw new Error('The provided double value is non-finite')
          return (
            this.seekDeferred && (this.seekDeferred = null),
            this.freeStore.eventAtom.setActionCaller('iSeek', i),
            (this.seekDeferred = this.rootPlayer.addDispose(new se.B())),
            this.mediaStore.setState({
              seekLoad: !0,
            }),
            this.rootPlayer.emit(
              a.t.Player_ProgressBar_Change,
              new CustomEvent(a.t.Player_ProgressBar_Change, {
                detail: {
                  progressTime: c,
                  initiator: i,
                },
              }),
            ),
            this.mediaStore.internalSeek(c),
            this.dp.seekAction && this.dp.seekAction(),
            (this.dp.seekAction = (0, o.gx)(
              function () {
                return !y.mediaStore.state.seeking && !y.mediaStore.video.seeking
              },
              function () {
                return y.seekDeferred.resolve()
              },
            )),
            [4, this.seekDeferred.promise]
          )
        case 1:
          return m.sent(), null != s.autoplay ? [2, s.autoplay ? this.play(t) : this.pause(t)] : [2]
      }
    })
  })
}

```

**Reply by HornyPrivateGamer** [reply]
> 脚本调的也是 b 站提供的播放器 API (#4341) 以前调用 `seek` 的时候第二个参数就是是否暂停, 那时候是可以做到的, 不过现在这个参数改成一个叫 `initiator` 的东西了, 没法控制了
> 
> 附 b 站播放器 `seek` 的代码:
> 
> ```js
> t.prototype.seek = function (e, t) {
>   return (0, n.mG)(this, void 0, void 0, function () {
>     var r,
>       i,
>       s,
>       u,
>       c,
>       l,
>       d,
>       f,
>       p,
>       h,
>       y = this
>     return (0, n.Jh)(this, function (m) {
>       switch (m.label) {
>         case 0:
>           if (
>             ((r = this.rootPlayer.disabled.seek || this.rootPlayer.hasFeature('noSeek')),
>             this.log.i(
>               '[seek]',
>               ee(
>                 ve ||
>                   (ve = (0, n.cy)(
>                     ['params:', ',alt:', ',disabled:', ''],
>                     ['params:', ',alt:', ',disabled:', ''],
>                   )),
>                 e,
>                 t,
>                 r,
>               ),
>             ),
>             (i = (null == t ? void 0 : t.initiator) || null),
>             (u = (s =
>               'number' == typeof e
>                 ? {
>                     value: e,
>                   }
>                 : e).isOffset
>               ? this.getCurrentTime() + s.value
>               : s.value),
>             (c = (0, C.Z)(u, 0, this.getDuration() - 0.1)),
>             this.patchAdvStore.hasValidAdv)
>           ) {
>             if (
>               ((l = (0, C.Z)(e, 0, this.getDuration() - 0.1)),
>               (d = this.patchAdvStore.getSeekDecision(l)),
>               (f = d.isSameEp),
>               (p = d.t),
>               (h = d.index),
>               !f)
>             )
>               return (
>                 this.patchAdvStore.setState({
>                   index: h,
>                 }),
>                 [2, this.patchAdvStore.patchSwitch(p)]
>               )
>             c = p
>           }
>           if (r) throw new Error('Seek Disabled')
>           if (isNaN(c)) throw new Error('The provided double value is non-finite')
>           return (
>             this.seekDeferred && (this.seekDeferred = null),
>             this.freeStore.eventAtom.setActionCaller('iSeek', i),
>             (this.seekDeferred = this.rootPlayer.addDispose(new se.B())),
>             this.mediaStore.setState({
>               seekLoad: !0,
>             }),
>             this.rootPlayer.emit(
>               a.t.Player_ProgressBar_Change,
>               new CustomEvent(a.t.Player_ProgressBar_Change, {
>                 detail: {
>                   progressTime: c,
>                   initiator: i,
>                 },
>               }),
>             ),
>             this.mediaStore.internalSeek(c),
>             this.dp.seekAction && this.dp.seekAction(),
>             (this.dp.seekAction = (0, o.gx)(
>               function () {
>                 return !y.mediaStore.state.seeking && !y.mediaStore.video.seeking
>               },
>               function () {
>                 return y.seekDeferred.resolve()
>               },
>             )),
>             [4, this.seekDeferred.promise]
>           )
>         case 1:
>           return m.sent(), null != s.autoplay ? [2, s.autoplay ? this.play(t) : this.pause(t)] : [2]
>       }
>     })
>   })
> }
> ```

所以目前是不可能的吧，对不对？

不懂编程，给点主意: 不能把initiator 改成以前那个参数吗？. 

真的不知道他们为什么不给个选择 :(

麻烦了，谢谢

**Comment by HornyPrivateGamer** [comment]
@the1812 
而且我又想到一个主意. 
你可以弄一个程序, 当暂停快进的时候自动暂停,那不就好了.
刚刚好对冲一下
那边要继续播放，你这边暂停一下. 
等于没有播放

还是老样子 如果楼主同意做这个功能的话，我立刻可以给100， 事成之后再给150
谢谢

**Reply by the1812** [reply]
冲不掉的，你不知道 b 站播放的时机，他可能比我们触发暂停早也可能晚，晚的时候照样播放

**Comment by YingDoge** [comment]
非常赞同，早就想说了. 全世界都是暂停的时候快进视频，不会播放就B在搞特殊

**Reply by Geeyun-JY3** [reply]
别尬黑，今天实测爱奇艺、腾讯视频和 AcFun 暂停的时候快进都会变成播放。
